### ISSUE TITLE
Unable to sync testnet (bitcoin or bitcoincash)

### ISSUE THREAD

**Issue by fireduck64** [kyuupichan/electrumx#545]
I suspect I am doing something dumb, but I can't figure it out.  Running most recent from master right now.

```
INFO:Controller:Python version: 3.6.2 (default, Aug  4 2017, 14:35:04) 
[GCC 6.4.0 20170724]
INFO:Controller:software version: ElectrumX 1.7.2
INFO:Controller:aiorpcX version: 0.5.9
INFO:Controller:supported protocol versions: 1.1-1.4
INFO:Controller:event loop policy: None
INFO:Controller:reorg limit is 8,000 blocks
INFO:Daemon:daemon #1 at bitd-test-abc.1209k.com:18332/ (current)
INFO:BlockProcessor:switching current directory to /data/elex
INFO:BlockProcessor:using rocksdb for DB backend
INFO:BlockProcessor:opened UTXO DB (for sync: True)
INFO:BlockProcessor:DB version: 6
INFO:BlockProcessor:coin: BitcoinCash
INFO:BlockProcessor:network: testnet
INFO:BlockProcessor:height: 0
INFO:BlockProcessor:tip: 000000000933ea01ad0ee984209779baaec3ced90fa3f408719526f8d77f4943
INFO:BlockProcessor:tx count: 0
INFO:History:history DB version: 0
INFO:History:flush count: 1
INFO:BlockProcessor:flushing DB cache at 1,200 MB
INFO:SessionManager:RPC server listening on localhost:8000
INFO:Prefetcher:catching up to daemon height 1,249,545 (1,249,545 blocks behind)
ERROR:Daemon:disconnected.  Retrying occasionally...
```
That last like of retrying occasionally continues for ever.

Here is my startup and environment script:
```
export DB_DIRECTORY=/data/elex

export DAEMON_URL=http://secretsomitted@bitd-test-abc.1209k.com:18332/
export TCP_PORT=50001
export SSL_PORT=50002
export HOST=""
export REPORT_HOST="electrumx-test.1209k.com"
export BANNER_FILE=/home/nerd/electrumx.git/banner

export SSL_CERTFILE=/home/nerd/electrumx.git/server.crt
export SSL_KEYFILE=/home/nerd/electrumx.git/server.key

export DB_ENGINE=rocksdb
ulimit -n 8192

export COIN=BitcoinCash
export NET=testnet

python3.6 ./electrumx_server
```

**Comment by fireduck64** [comment]
on the bitcoind side:
2018-07-31 17:04:27 connection from [2a01:4f8:212:3b1f::2]:31384 dropped (banned)
2018-07-31 17:06:31 connection from [2a01:4f8:212:3b1f::2]:39683 dropped (banned)

Might be a problem there.  Investigating.

**Comment by fireduck64** [comment]
Nevermind, those banned nodes have nothing to do with me.

**Comment by fireduck64** [comment]
I've tried various tags all the way back to 1.3 and still no syncing with the same results.

**Comment by kyuupichan** [comment]
I have no idea - it's some issue connecting to your daemon right - it's just not working.  I don't think it's anything in ElectrumX code, more likely the network / firewall rules you have set up, or possibly the bitcoind setup itself.

**Comment by kyuupichan** [comment]
Well, numerous people run it without issue, and all past daemon connection problems have been in the configuration somewhere.  I guess some package like aiohttp might be on a problematic revision.

Regarding logging, I don't think so - I suspect in daemon.py it's just getting an exception from aiohttp when the connection closes

**Comment by kyuupichan** [comment]
It's weird though - it's got the daemon height, so some communication is happening and succeeding... do you have some kind of HTTP workqueue limit set in bitcoind?  Try setting the semaphore initial value in daemon.py line 45 to 1

**Comment by fireduck64** [comment]
My best guess is that something about there being a million blocks causes
too many requests and the bitcoind disconnecting the client or timing out.
I'll try the line you mentioned when I can.

**Comment by kyuupichan** [comment]
Well it's not supposed to be that way - that's the point of the semaphore.  It seems to be right at the start - it only requests 10 blocks initially - so it's all very odd

**Comment by fireduck64** [comment]
Very strange, if I put a plain TCP relay service (my session recorder for debugging things) it works perfectly.

Really strange.

**Comment by fireduck64** [comment]
Ok, I have something.  If both the electrumx server and the bitcoind are on 9000 mtu network, then this problem comes up.  If the MTU is 1500, then it doesn't happen.

**Comment by kyuupichan** [comment]
@fireduck64 any more info?

**Comment by fireduck64** [comment]
Yeah, I screwed up and I'm sorry for wasting your time. ;)

I was switching my local network from 1500 to 9000 mtu.  In my testing it seemed that mixed mtu things would figure it out and work fine (throw a few icmp must frags at each other or whatever).  However, I didn't reboot everything and there was an odd interaction between the mtu on the VM server host OS and the child VMs and I actually only rebooted the bitcoin test VMs, so they were on the 9000 mtu and unable to communicate large packets.  So I thought it was a testnet problem, not a VM networking problem.

So it exhibited as the systems could connect, do a little communication but as soon as they started doing large queries the TCP sessions stalled and died.
